src/http.rs

```diff
@@ -53,50 +52,52 @@ async fn run(
 
     // GET /hd/:fingerprint
     let hd_wallet_handler = warp::get()
-        .and(warp::path!("hd" / Fingerprint))
+        .and(warp::path!("hd" / DescriptorChecksum))
         .and(query.clone())
-        .map(|fingerprint: Fingerprint, query: Arc<Query>| {
+        .map(|desc_check: DescriptorChecksum, query: Arc<Query>| {
             let wallet = query
-                .get_hd_wallet(fingerprint)
+                .get_hd_wallet(desc_check)
                 .or_err(StatusCode::NOT_FOUND)?;
             Ok(reply::json(&wallet))
         })
         .map(handle_error);
 
     // GET /hd/:fingerprint/:index
     let hd_key_handler = warp::get()
-        .and(warp::path!("hd" / Fingerprint / u32))
+        .and(warp::path!("hd" / DescriptorChecksum / u32))
         .and(query.clone())
-        .map(|fingerprint: Fingerprint, index: u32, query: Arc<Query>| {
-            let script_info = query
-                .get_hd_script_info(fingerprint, index)
-                .or_err(StatusCode::NOT_FOUND)?;
-            Ok(reply::json(&script_info))
-        })
+        .map(
+            |desc_check: DescriptorChecksum, index: u32, query: Arc<Query>| {
+                let script_info = query
+                    .get_hd_script_info(desc_check, index)
+                    .or_err(StatusCode::NOT_FOUND)?;
+                Ok(reply::json(&script_info))
+            },
+        )
         .map(handle_error);
 
     // GET /hd/:fingerprint/gap
     let hd_gap_handler = warp::get()
-        .and(warp::path!("hd" / Fingerprint / "gap"))
+        .and(warp::path!("hd" / DescriptorChecksum / "gap"))
         .and(query.clone())
-        .map(|fingerprint: Fingerprint, query: Arc<Query>| {
+        .map(|desc_check: DescriptorChecksum, query: Arc<Query>| {
             let gap = query
-                .find_hd_gap(fingerprint)
+                .find_hd_gap(desc_check)
                 .or_err(StatusCode::NOT_FOUND)?;
             Ok(reply::json(&gap))
         })
         .map(handle_error);
 
     // GET /hd/:fingerprint/next
     let hd_next_handler = warp::get()
-        .and(warp::path!("hd" / Fingerprint / "next"))
+        .and(warp::path!("hd" / DescriptorChecksum / "next"))
         .and(query.clone())
-        .map(|fingerprint: Fingerprint, query: Arc<Query>| {
+        .map(|desc_check: DescriptorChecksum, query: Arc<Query>| {
             let wallet = query
-                .get_hd_wallet(fingerprint)
+                .get_hd_wallet(desc_check.clone())
```

Comment:
`DescriptorChecksum` is small enough to implement `Copy`, so that cloning won't be necessary.

Reply:
But `DescriptorChecksum` contains a `string` so we can't derive `Copy` ... I'm going to leave this one to you ...